src/test/fuzz/prevector.cpp

```diff
@@ -0,0 +1,268 @@
+// Copyright (c) 2015-2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+
+#include <vector>
+#include <prevector.h>
+
+#include <reverse_iterator.h>
+#include <serialize.h>
+#include <streams.h>
+
+namespace {
+
+template<unsigned int N, typename T>
+class prevector_tester {
+    typedef std::vector<T> realtype;
+    realtype real_vector;
+    realtype real_vector_alt;
+
+    typedef prevector<N, T> pretype;
+    pretype pre_vector;
+    pretype pre_vector_alt;
+
+    typedef typename pretype::size_type Size;
+
+public:
+    void test() {
```

Comment:
Can the test method be made read-only? If yes, I think `case 15` can be removed?

```suggestion
    void test() const {
```

Reply:
Right, my thinking was that the fuzzer should be allowed to find intermediate failures too, but I think that's pointless. A shorter fuzzer input (which runs test once at the end) would accomplish the same.

Done.